src/orion/core/utils/working_dir.py

```diff
@@ -24,16 +24,22 @@ def __init__(self, working_dir, temp=True, suffix=None, prefix=None):
             Name of the directory. If empty, will create a temporary one.
 
         """
-        self.working_dir = working_dir
         self._temp = temp
         self._suffix = suffix
         self._prefix = prefix
         self._tmpdir = None
 
+        if working_dir:
+            self.working_dir = os.path.abspath(working_dir)
+        else:
+            self.working_dir = os.path.join(tempfile.gettempdir(), 'orion')
```

Comment:
:thinking: Maybe leave the if-else in the consumer. The `os.makedirs` can be remove entirely as the dir will be created in `__enter__` anyway.

Reply:
I didn't need to move it then ;)

Reply:
Yes, my comment was only about `os.makedirs` and it was about **re**moving it. :) (https://github.com/Epistimio/orion/pull/184#discussion_r295933318)

Reply:
Semantics :roll_eyes: